File: 1.0/src/c/cz/mg/vulkan/PFN_vkDestroyDebugUtilsMessengerEXT.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkDestroyDebugUtilsMessengerEXT_callNative(JNIEnv* env, jclass jc, jlong address, jlong instance, jlong messenger, jlong pAllocator){
    (void)env;
    (void)jc;
    PFN_vkDestroyDebugUtilsMessengerEXT f = (PFN_vkDestroyDebugUtilsMessengerEXT)jniLongToFunctionPointer(address);
    f(
        *((VkInstance*)jniLongToPointer(instance)),
        *((VkDebugUtilsMessengerEXT*)jniLongToPointer(messenger)),
        ((VkAllocationCallbacks*)jniLongToPointer(pAllocator))
    );
}

File: 1.0/src/cz/mg/vulkan/PFNvkDestroyDebugUtilsMessengerEXT.java
package cz.mg.vulkan;

public class PFNvkDestroyDebugUtilsMessengerEXT extends VkFunctionPointer {
    public PFNvkDestroyDebugUtilsMessengerEXT() {
    }

    protected PFNvkDestroyDebugUtilsMessengerEXT(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkDestroyDebugUtilsMessengerEXT(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkDestroyDebugUtilsMessengerEXT(long value) {
        setValue(value);
    }

    public PFNvkDestroyDebugUtilsMessengerEXT(VkInstance instance) {
        super(instance, new VkString("vkDestroyDebugUtilsMessengerEXT"));
    }

    public void call(VkInstance instance, VkDebugUtilsMessengerEXT messenger, VkAllocationCallbacks pAllocator){
        callNative(getValue(), instance != null ? instance.getVkAddress() : VkPointer.getNullAddressNative(), messenger != null ? messenger.getVkAddress() : VkPointer.getNullAddressNative(), pAllocator != null ? pAllocator.getVkAddress() : VkPointer.NULL);
    }

    protected static native void callNative(long vkaddress, long instance, long messenger, long pAllocator);



}
